usautobuild/config.py

```diff
@@ -91,36 +95,31 @@ def handle_args(self):
         self.set_config(config)
 
     def set_config(self, config):
-        if config.get("build_number"):
-            self.build_number = config.get("build_number")
-        if config.get("git_url"):
-            self.git_url = config.get("git_url")
-        if config.get("git_branch"):
-            self.git_branch = config.get("git_branch")
-        if config.get("unity_version"):
-            self.unity_version = config.get("unity_version")
-        if config.get("target_platforms"):
-            self.target_platforms = config.get("target_platforms")
-        if config.get("cdn_download_url"):
-            self.cdn_download_url = config.get("cdn_download_url")
-        if config.get("forkname"):
-            self.forkname = config.get("forkname")
-        if config.get("output_dir"):
-            self.output_dir = config.get("output_dir")
-        if config.get("license_file"):
-            self.license_file = config.get("license_file")
-        if config.get("abort_build_on_fail"):
-            self.abort_on_build_fail = config.get("abort_build_on_fail")
-        if config.get("discord_webhook"):
-            self.discord_webhook = config.get("discord_webhook")
-        if config.get("allow_new_changes"):
-            self.allow_no_changes = config.get("allow_new_changes")
-        if config.get("abort_on_build_fail"):
-            self.abort_on_build_fail = config.get("abort_on_build_fail")
+        for path_var in (
+            "output_dir",
+            "license_file",
+        ):
+            if path_var in config:
+                setattr(self, path_var, Path(config[path_var]))
+
```

Comment:
What is setattr?

Reply:
https://docs.python.org/3/library/functions.html#setattr

an object's dot `.` with assignment `=` operation just with string variable as a name. there is a group of similar methods: getattr, setattr, delattr.
these should only be used in cases where you have no other choice (like here, prevents huge code duplication) because they are not type checker friendly / less readable overall